stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2862,6 +2862,29 @@ impl NakamotoChainState {
         Self::get_block_header_nakamoto(chainstate_conn.sqlite(), &block_id)
     }
 
+    /// Get all the blocks for a specific consensus hash and burnview
+    /// highest block on top
+    pub fn get_block_headers_in_tenure_at_burnview(
+        db: &Connection,
+        tenure_id: &ConsensusHash,
+        burn_view: &ConsensusHash,
+    ) -> Result<Vec<StacksHeaderInfo>, ChainstateError> {
```

Comment:
This code is not fork-aware and will need to be rewritten.  Just flagging this for the review (I know you already know and are working on it :slightly_smiling_face:).

Reply:
Moved the logic into the stream generator (a new block is processed at each chunk).